packages/superset-ui-core/src/query/normalizeOrderBy.ts

```diff
@@ -19,46 +19,50 @@
 import isEmpty from 'lodash/isEmpty';
 import isBoolean from 'lodash/isBoolean';
 
-import { QueryObject } from './types';
+import { FormDataResidual, QueryFormOrderBy } from './types';
+import { t } from '../translation';
 
-export default function normalizeOrderBy(queryObject: QueryObject): QueryObject {
-  if (Array.isArray(queryObject.orderby) && queryObject.orderby.length > 0) {
+const defaultOrderby = undefined;
+
+export default function normalizeOrderBy(
+  formData: FormDataResidual,
+): QueryFormOrderBy[] | typeof defaultOrderby {
+  if (Array.isArray(formData.orderby) && formData.orderby.length > 0) {
+    const orderbyClauses = formData.orderby.map(item => {
+      // value can be in the format of `['["col1", true]', '["col2", false]']`,
+      // where the option strings come directly from `order_by_choices`.
+      if (typeof item === 'string') {
+        try {
+          return JSON.parse(item);
+        } catch (error) {
+          throw new Error(t('Found invalid orderby options'));
+        }
+      }
```

Comment:
Why are we `JSON.parse`ing a `string` object here?

Reply:
The item my be a JSON payload for orderby
```
['["col1", true]', '["col2", false]']
```

Reply:
Oh now I noticed you were just moving this code unchanged... I wonder if this is just for backwards compatibility with some old version of the control, or if it's still actively passing JSON formatted strings. If it's still doing that, we should probably update the type here: https://github.com/apache-superset/superset-ui/blob/8249e92f0d94fd6f7792121ed2354631d9532aee/packages/superset-ui-chart-controls/src/types.ts#L54 Or better yet, make sure it actually passes `[string, string][] | null`, not `str[] | null` 😆